### DISCUSSION TITLE
USB Pinout for controllers?

### DISCUSSION THREAD

**Discussion by Helder1981** [nand2mario/nestang#29] [General]
I saw some info in the closed issues and it showed player 1 usb port but this info is nowhere else in the github so I would like to know if this is indeed the correct pinout and also what would be Player 2 usb port pinout and if this can be added to the wiki so it's easily found by others.

Quote from Closed Issues post:

```
 Connect D+ to pin 41 and D- to 42, and add two 15k pull-down resistors as shown below.
```

<img width="287" alt="usb_softphy" src="https://github.com/nand2mario/nestang/assets/113511414/960e8dfb-b0e9-42ce-8f4b-09912562c564">

My plan is to make a simple pcb that will have a usb port with the necessary resistors. I'll gladly put it on the github here for all to get manufactured once I am sure of the pinout.

**Comment by nand2mario** [comment]
The pins are documented [here](https://github.com/nand2mario/nestang/blob/master/doc/usb_gamepad.md).

```
For controller 1, connect D+ to pin 42 and D- to pin 41.
For controller 2, connect D+ to pin 56 and D- to pin 54.
```

How about making it a hat board with two usb ports, stacking on top of the nano? A bit like the MiSTer IO board.

**Reply by Helder1981** [reply]
Sure thing, I can also add the PS2 controller ports to this pcb, even have the esp32 attach to it to do the blueretro which I think is something to implement going forward since it's not an expensive addon and will open up tons of controllers. I am waiting for my nano to come in so I can design things properly.

**Comment by Helder1981** [comment]
Just finished designing the board and ordering a few, it will have usb ports in the middle of the PS2 controllers as well and a place for the ESP32 NodeMCU-32S board for doing BlueRetro. Once I get them in (roughly 2 weeks or so) I will confirm if they are good and working and if they are I will share the files and sell off the extras I ordered.
![Screenshot_](https://github.com/nand2mario/nestang/assets/8185885/d287c254-cb17-45d5-80b7-ee3064dd2e99)

**Reply by nand2mario** [reply]
Looks great!
